toplevel/identity/entity.go

```diff
@@ -0,0 +1,642 @@
+package identity
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/app-sre/vault-manager/pkg/utils"
+	"github.com/app-sre/vault-manager/pkg/vault"
+	"github.com/app-sre/vault-manager/toplevel"
+	"github.com/hashicorp/go-version"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+)
+
+type config struct{}
+
+var _ toplevel.Configuration = config{}
+
+type user struct {
+	Name        string `yaml:"name"`
+	OrgUsername string `yaml:"org_username"`
+	Roles       []role `yaml:"roles"`
+}
+
+type role struct {
+	Name        string           `yaml:"name"`
+	Permissions []oidcPermission `yaml:"oidc_permissions"`
+}
+
+type oidcPermission struct {
+	Name        string        `yaml:"name"`
+	Description string        `yaml:"description"`
+	Service     string        `yaml:"service"`
+	Policies    []vaultPolicy `yaml:"vault_policies"`
+}
+
+type vaultPolicy struct {
+	Name string `yaml:"name"`
+}
+
+type entity struct {
+	Name     string
+	Id       string
+	Type     string
+	Metadata interface{}
+	Policies []string
+	Aliases  []entityAlias
+}
+
+type entityAlias struct {
+	Name           string
+	Id             string
+	Type           string
+	AuthType       string
+	AccessorId     string
+	CustomMetadata interface{}
+}
+
+var _ vault.Item = entity{}
```

Comment:
what is the purpose of this?

Reply:
not saying its wrong I just don't know this pattern immediately off the top of my head

Reply:
it was a new pattern to me as well but a practice employed by the other integrations so I opted to include. It is a "check" that occurs during compile time to see if the `entity` structure implements the `vault.Item` interface correctly.